module/EntraBeta/Microsoft.Entra.Beta/Applications/Grant-EntraBetaMCPServerPermission.ps1

```
@@ -0,0 +1,208 @@
+        [Parameter(ParameterSetName = 'CustomClientScopes', Mandatory = $true, HelpMessage = "Specify one or more specific scopes to grant to the MCP client. If not specified, all available scopes will be granted.")]
+        [ValidateNotNullOrEmpty()]
+        [string[]]$Scopes
+    )
+
+    begin {
+        # Create headers for the request
+        $customHeaders = New-EntraBetaCustomHeaders -Command $MyInvocation.MyCommand
+
+        # Ensure connection to Microsoft Entra
+        if (-not (Get-EntraContext)) {
+            $errorMessage = "Not connected to Microsoft Graph. Use 'Connect-Entra -Scopes Application.ReadWrite.All, Directory.Read.All, DelegatedPermissionGrant.ReadWrite.All, ' to authenticate."
+            Write-Error -Message $errorMessage -ErrorAction Stop
+            return
+        }
+
+        function Get-ServicePrincipal([string]$appId, [string]$name) {
+            $sp = Get-MgBetaServicePrincipal -Filter "appId eq '$appId'" -ErrorAction SilentlyContinue -Headers $customHeaders | Select-Object -First 1
+            if (-not $sp) {
+                Write-Verbose "Creating service principal for $name ..."
+                $sp = New-MgBetaServicePrincipal -AppId $appId -Headers $customHeaders
+            }
+            return $sp
+        }
+
+        function Get-Grant {
+            param(
+                [Parameter(Mandatory)] [string] $ClientSpId,
+                [Parameter(Mandatory)] [string] $ResourceSpId
+            )
+            Get-MgBetaOauth2PermissionGrant `
+                -Filter "clientId eq '$ClientSpId' and resourceId eq '$ResourceSpId' and consentType eq 'AllPrincipals'" `
+                -Top 1 `
+                -Property "id,scope,clientId,resourceId,consentType" `
+                -ErrorAction SilentlyContinue `
```

Why silently continue?